### PR TITLE
Cmake revs

### PR DESCRIPTION
**Pull request by meshula**
- Consolidate build settings, and cmake options to the top level file
- add option for shared or static libs
- add "d' suffix as is required for debug python builds on windows
- more granular build control (turn python builds or c++ builds on or off independently)
- For IDEs, rename the .sln or .xcode file to OpenTimelineIO (was OPENTIMELINEIO_ROOT)
- macos: Make Rpath respect the specified python installation directory

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,141 @@
...
 
+set(OTIO_AUTHOR "Contributors to the OpenTimelineIO project")
+set(OTIO_AUTHOR_EMAIL "opentimelineio@pixar.com")
+set(OTIO_LICENSE "Modified Apache 2.0 License")
+
+project(OpenTimelineIO VERSION ${OTIO_VERSION} LANGUAGES C CXX)
+
+#------------------------------------------------------------------------------
+# Options
+
```
Comment (ssteinbach) [inline CMakeLists.txt:24]: can you add a note that the idea with this section is that any `option()` calls in the cmake system be put here and only here

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,141 @@
...
+option(OTIO_CXX_COVERAGE "Invoke gcov if available" OFF)
```
Comment (meshula) [inline CMakeLists.txt:25]: add a note about code coverage using gcov

Comment (jminor) [inline CMakeLists.txt:25]: WBN to mention "code coverage" in case people don't know what gcov is.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,10 +1,141 @@
...
+message(INFO, ${OTIO_PYTHON_INSTALL_DIR})
+
+if(OTIO_SHARED_LIBS)
+    message(INFO, "Building shared libs....")
+    set(OTIO_SHARED_OR_STATIC_LIB "SHARED")
+else()
+    message(INFO, "Building static libs....")
+    set(OTIO_SHARED_OR_STATIC_LIB "STATIC")
+    if (OTIO_PYTHON_ISTALL AND NOT MSVC)
+        # A revision of OTIO will include explicit visibility decorations to resolve this issue
```
Comment (ssteinbach) [inline CMakeLists.txt:42]: this should be a TODO

**Comment by meshula** [comment]
Addresses: https://github.com/PixarAnimationStudios/OpenTimelineIO/issues/666

**Comment by meshula** [comment]
Closing in favor of #837

**Pull request closed without merging**